consensus/istanbul/core/commit.go

```diff
@@ -207,62 +218,135 @@ func (c *core) handleCheckedCommitForCurrentSequence(msg *istanbul.Message, comm
 		return errInvalidValidatorAddress
 	}
 
-	if err := c.verifyCommittedSeal(commit, validator); err != nil {
-		return errInvalidCommittedSeal
+	// ensure that the commit is for the current proposal
+	if err := c.verifyCommit(commit); err != nil {
+		return err
 	}
 
 	newValSet, err := c.backend.NextBlockValidators(c.current.Proposal())
 	if err != nil {
 		return err
 	}
 
+	// Verifies the individual seal for this commit message for the epoch
+	// block, no-op unless this is the last block of an epoch.
```

Comment:
We should optimize this to use the same aggregation techniques.

Reply:
Can do, I think It could be a significant amount of work, do we want to make it part of this PR?

Reply:
I think so yes, because without that work we will suffer (further) issues on epoch blocks if we raise the number of validators

Reply:
This is now done